src/oph/ehoks/palaute/opiskelija.clj

```diff
@@ -128,22 +104,43 @@
   ([{:keys [hoks opiskeluoikeus] :as ctx} kysely-type opts]
     (jdbc/with-db-transaction
       [tx db/spec]
-      (let [ctx     (assoc ctx
-                           :tx              tx
-                           :koulutustoimija (palaute/koulutustoimija-oid!
+      (let [ctx (assoc ctx :koulutustoimija (palaute/koulutustoimija-oid!
                                               opiskeluoikeus))
-            palaute {:type kysely-type
-                     :existing-heratteet
-                     (when-not (:resend? opts)
-                       (existing-heratteet! ctx kysely-type))}
-            [state field reason] (initial-palaute-state-and-reason ctx palaute)
-            tapahtuma {:reason     reason
-                       :other-info (select-keys hoks [field])}]
+            existing-heratteet (when-not (:resend? opts)
+                                 (existing-heratteet! tx ctx kysely-type))
+            [state field reason] (initial-palaute-state-and-reason
+                                   ctx kysely-type existing-heratteet)]
         (log/info "Initial state for" kysely-type "for HOKS" (:id hoks)
                   "will be" (or state :ei-luoda-ollenkaan)
                   "because of" reason "in" field)
         (when state
-          (initiate! ctx (assoc palaute :state state :tapahtuma tapahtuma)))
+          (let [heratepvm       (get hoks (herate-date-basis kysely-type))
+                existing-herate (find-first palaute/unhandled?
+                                            existing-heratteet)
+                target-kasittelytila (not= state :odottaa-kasittelya)
+                amisherate-kasittelytila
+                (db-hoks/get-or-create-amisherate-kasittelytila-by-hoks-id!
+                  (:id hoks))]
+            (db-hoks/update-amisherate-kasittelytilat!
+              tx {:id (:id amisherate-kasittelytila)
+                  (kysely-kasittely-field-mapping kysely-type)
+                  target-kasittelytila})
+            (->> (build! ctx kysely-type state existing-herate)
+                 (palaute/upsert! tx)
+                 (palautetapahtuma/build ctx state field reason existing-herate)
+                 (palautetapahtuma/insert! tx))
+            (when (= :odottaa-kasittelya state)
```

Comment:
tää `target-kasittelytila`:n perusteella tai toisin päin, pitää myös miettiä kyseisen muuttujan nimeä
vakavuus: ehdotus